react-app/App.jsx

```diff
@@ -3,28 +3,31 @@ import DropZoneFile from '@components/DropZoneFile';
 import ListImages from '@components/ListImages';
 import Gallery from '@components/Gallery';
 import { GalleryContext } from '@contexts/galleryContext';
+import globalObject from '@constants/globalObject';
+import { getItemLocalStorage } from '@utils/localStorageUtils';
 
 import './App.css';
 
 const App = () => {
-  const [isGalleryOpened, setIsGalleryOpened] = useState(false);
   const [filesToUploadList, setFilesToUploadList] = useState([]);
+  const { xblockId, isStudioView } = globalObject;
 
   useEffect(() => {
-    localStorage.clear();
+    const filesUploaded = getItemLocalStorage(xblockId) || [];
+    setFilesToUploadList(filesUploaded);
```

Comment:
This is good enough for now, but remember we should remove this once we save the list as part of the xblock state.